coral/src/app/features/topics/acl-request/forms/TopicConsumerForm.tsx

```diff
@@ -70,7 +71,27 @@ const TopicConsumerForm = ({
   }, [aclIpPrincipleType]);
 
   const { mutate, isLoading, isError, error } = useMutation({
-    mutationFn: createAclRequest,
+    mutationFn: async (payload: TopicConsumerFormSchema) => {
+      const { teamId, error } = await getTopicTeam({
+        topicName: payload.topicname,
+        patternType: payload.aclPatternType,
+      });
+
+      // teamId will actually never be undefined, but will be 0 when there is an error
+      // However, the openapi.yaml definition doesn't reflect that, so we need to check for undefined
+      // So that teamId in the happy path can be typed properly
+      if (error !== undefined || teamId === undefined) {
+        const errorMessage =
```

Comment:
1. Thanks for the comments. I've seen the slack conversation, but it would have taken me a moment to remember that and figure out why there's the check and all ❤️ 

2. The comment says `teamId` will be `0` if there is an error. But we're not checking for that?

Reply:
> The comment says teamId will be 0 if there is an error. But we're not checking for that?

Yes, it is unnecessary, `teamId` is only `0` when `error` is not `undefined` (it's the "error" value of `teamId`). I think I should add it to the comments! Also will create an issue to fix the behaviour of the `getTopicTeam` endpoint so we don't forget (even if @aindriu-aiven is aware, it doesn't hurt to document it ^^)

Reply:
https://github.com/Aiven-Open/klaw/issues/1710

Reply:
[f61beb6](https://github.com/Aiven-Open/klaw/pull/1702/commits/f61beb6d08ceca7e5384f529bc027a930a75624f)

Reply:
💡  Yes, I understand - and the comment is more clear now, too. Thanks!